push_notification_test.go

```diff
@@ -152,5 +170,197 @@ func TestPushNotifications(t *testing.T) {
 				})
 			})
 		})
+
+		Convey("when authenticating a User", func() {
+			Convey("should return an error if the User Id is empty", func() {
+				token, err := pn.AuthenticateUser("")
+
+				So(err, ShouldNotBeNil)
+				So(token, ShouldEqual, "")
+				So(err.Error(), ShouldContainSubstring, "User Id cannot be empty")
+			})
+
+			Convey("should return an error if the User Id is too long", func() {
+				s := ""
+				for i := 0; i < maxUserIdLength; i++ {
+					s += "a"
+				}
+
+				token, err := pn.AuthenticateUser(s)
+
+				So(err, ShouldBeNil)
+				So(token, ShouldNotEqual, "")
+
+				longerUserId := s + "a"
+				token, err = pn.AuthenticateUser(s + "a")
+
+				So(err, ShouldNotBeNil)
+				So(token, ShouldEqual, "")
+				So(
+					err.Error(),
+					ShouldContainSubstring,
+					fmt.Sprintf("User Id ('%s') length too long (expected fewer than %d characters, got %d)", longerUserId, maxUserIdLength+1, len(longerUserId)),
+				)
+			})
+
+			Convey("should return a valid JWT token if everything is correct", func() {
+				token, err := pn.AuthenticateUser("u-123")
+				expectedIssuer := "https://" + testInstanceId + ".pushnotifications.pusher.com"
+				expectedSubject := "u-123"
+
+				So(err, ShouldBeNil)
```

Comment:
I'd have these `err shouldBeNil` always right after calling a function as to not forget it about it.

I'd move these `expected` strings to where they are being used (principle of smallest variable scope).